sympy/liealgebras/cartan_base.py

```diff
@@ -0,0 +1,286 @@
+from sympy.core import Basic
+from sympy.matrices import zeros, Matrix, eye, ones
+import warnings
+
+class Standard_Cartan(Basic):
+    """
+    Semi-Concrete base class for Cartan types such as A4, etc. In this module we make
+    certain choices about how to represent the algebra.  This class is never meant to be
+    called directly. It is exported as `CartanType` and should be called that way.
+
+    Examples
+    ========
+    >>> from sympy.liealgebras import CartanType
+    >>> CartanType("G2")
+    TypeG('G', 2)
+
+    Sources
+    =======
+    - https://inspirehep.net/literature/181168
+    - https://www.researchgate.net/publication/252487471_Group_Theory_A_Physicist's_Survey
+    - https://arxiv.org/pdf/1206.6379.pdf
+    """
+
+    def __new__(cls, series, n):
+        obj = Basic.__new__(cls, series, n)
+        obj.n = n
+        obj.series = series
+        return obj
```

Comment:
What kinds of objects are series and n here? Basic instances should have Basic args. We normally need to use sympify for this. Also we should derive any attributes from the args treating obj itself as immutable.

The standard way to do this is:
```python
    def __new__(cls, series, n):
        series = _sympify(series)
        n = _sympify(n)
        return super().__new__(cls, series, n)

    @property
    def series(self):
        return self.args[0]

    @property
    def n(self):
        return self.args[1]
```
The `Basic.__new__` method will store the arguments in `args` for us. We only define `__new__` to make sure that the object is created with the right number of arguments, to sympify them and then possibly to check the type of the args.